drivers/adc/ad7779/ad7779.c

```diff
@@ -524,9 +552,16 @@ int32_t ad7779_get_gain(ad7779_dev *dev,
 			ad7779_ch ch,
 			ad7779_gain *gain)
 {
-	*gain = dev->gain[ch];
+	int ret = 0;
+	uint8_t reg_data;
 
-	return 0;
+	if (!dev->read_from_cache) {
+		ret = ad7779_spi_int_reg_read(dev, AD7779_REG_CH_CONFIG(ch), &reg_data);
```

Comment:
missing error check, there's no point in assigning a wrong value to the `gain` if the spi function failed. check throughout the entire code.

Reply:
Added ret check throughout